expr/functions/cairo/png/cairo.go

```diff
@@ -790,7 +791,7 @@ func EvalExprGraph(ctx context.Context, e parser.Expr, from, until int64, values
 					return nil, err
 				}
 				r.Dashed = d
-				r.Tags["dashed"] = fmt.Sprintf("%f", d)
+				r.Tags["dashed"] = strconv.FormatFloat(d, 'g', -1, 64)
```

Comment:
Move this (and conversion also) out of cycle. Also replace `r := *a` with CopyLink() for prevent no-copy-on write bug.

Reply:
Done. Note that I had to define `dashLen` and `dashLenStr` that are only used in the `"dashed"` case.  See https://github.com/go-graphite/carbonapi/pull/686/commits/1ecf628951964ebe99c77dd495be865304e6fb09